client/client_test.go

```diff
@@ -215,3 +215,37 @@ func TestGetActionTimeout(t *testing.T) {
 		})
 	}
 }
+
+func Test_argListFromMongoDBParams(t *testing.T) {
+	type testParams struct {
+		req      *agentpb.StartActionRequest_PTMongoDBSummaryParams
+		expected []string
+	}
+
+	testCases := []*testParams{
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "secret"}, []string{"--username", "person", "-psecret", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: ""}, []string{"--username", "person", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "",
+			Password: "secret"}, []string{"-psecret", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 65536, Username: "",
+			Password: "secret"}, []string{"-psecret", "10.20.30.40"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 555, Username: "", Password: "secret"},
+			[]string{"-psecret"}},
+
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 0, Username: "", Password: ""}, []string{}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 0, Username: "王华", Password: `"`},
+			[]string{"--username", "王华", `-p"`}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "   "}, []string{"--username", "person", "-p   ", "10.20.30.40:555"}},
```

Comment:
Sorry for being picky, can we test it with password = "has a blank" to check it is quoted ?

Reply:
It will not quote it. If we put there %q, the password does not work. It works if we quote it from the command line but it does not, if we pass it from pmm-agent.
In my opinion, the command line has to parse it and identify the params. If you pass it from one app to another, it is passed as an array of strings. So the destination app will receive the identical array of strings.